tests/AppBundle/MessageHandler/Delivery/DeliveryCreatedHandlerTest.php

```diff
@@ -0,0 +1,471 @@
+        $this->userManager = $this->prophesize(UserManager::class);
+        $this->emailManager = $this->prophesize(EmailManager::class);
+        $this->mjml = $this->prophesize(RendererInterface::class);
+        $this->messageBus = $this->prophesize(MessageBusInterface::class);
+        $this->translator = $this->prophesize(TranslatorInterface::class);
+        $this->twig = $this->prophesize(TwigEnvironment::class);
+        $this->settingsManager = $this->prophesize(SettingsManager::class);
+        $this->deliveryRepository = $this->prophesize(ObjectRepository::class);
+
+        $this->entityManager
+            ->getRepository(Delivery::class)
+            ->willReturn($this->deliveryRepository->reveal());
+
+        $this->userManager->findUsersByRoles(['ROLE_ADMIN', 'ROLE_DISPATCHER'])
+            ->willReturn([]);
+
+        $this->settingsManager->get('administrator_email')
+            ->willReturn(null);
+
+        $this->translator->trans('notifications.tap_to_open')
+            ->willReturn('Tap to open');
+
+        $this->handler = new DeliveryCreatedHandler(
+            $this->entityManager->reveal(),
+            $this->userManager->reveal(),
+            $this->emailManager->reveal(),
+            $this->mjml->reveal(),
+            $this->messageBus->reveal(),
+            $this->translator->reveal(),
+            $this->twig->reveal(),
+            $this->settingsManager->reveal(),
+            'en'
+        );
+    }
+
+    public function testSendDelivery__TYPE_SIMPLE__SamePickupAddr()
+    {
+        $pickup = new Task();
+        $pickup->setType(Task::TYPE_PICKUP);
+        $dropoff = new Task();
+        $dropoff->setType(Task::TYPE_DROPOFF);
+        $pickup->setNext($dropoff);
+        $dropoff->setPrevious($pickup);
+
+        $pickupAddress = new Address();
+        $pickupAddress->setStreetAddress("111 Nice Pickup St, Somewhere, Argentina");
+        $pickup->setAddress($pickupAddress);
+        $dropoffAddress = new Address();
+        $dropoffAddress->setStreetAddress("222 Nice Dropoff St, Someplace, Argentina");
+        $dropoff->setAddress($dropoffAddress);
+
+        $pickup->setAfter(new \DateTime('2025-01-02 01:02:03'));
+        $pickup->setBefore(new \DateTime('2025-01-02 02:03:04'));
+        $dropoff->setAfter(new \DateTime('2025-01-02 03:04:05'));
+        $dropoff->setBefore(new \DateTime('2025-01-02 04:05:06'));
+
+        $delivery = $this->prophesize(Delivery::class);
+        $delivery->getId()->willReturn(1);
+        $delivery->getTasks()->willReturn([$pickup, $dropoff]);
+        $delivery->getPickup()->willReturn($pickup);
+        $delivery->getDropoff()->willReturn($dropoff);
+
+        $order = $this->prophesize(Order::class);
+        $order->getId()->willReturn(11);
+        $order->getDelivery()->willReturn($delivery);
+        $delivery->getOrder()->willReturn($order);
+
+        /////////////////////////
+        // Test with a pickup address being the owner/store address
```

Comment:
I think it would be helpful to hard-code title as well

Reply:
Done at https://github.com/coopcycle/coopcycle-web/pull/5062/commits/18eb8c540b292ec16f9e5890399429fa978e56d9